### ISSUE TITLE
Crusader48's Feedback

### ISSUE THREAD

**Issue by finnbear** [SoftbearStudios/mk48#190]
I have suggested many things before by talking to the server in the mk48.io game, but this seems like a better place to mention them. There are multiple ships that I would like to see added to the game. These include the Skjold class corvette, the Nimitz or Gerald R Ford carrier, the Typhoon submarine, the USS Iowa (modernized), a minesweeper such as the HMCS Kingston, a hospital ship, an electronic countermeasures ship (ECM) and maybe a fishing trawler that catches fish to get xp. The carrier, submarine and Iowa would make a great lvl 10 and the skjold class would probably fit in at lvl 5 or so. I really think that the skjold would be cool because it travels 60kn, has 8 naval strike missiles, a gun and soft kill countermeasures as well as stealth capabilities. That’s pretty much it for ships, and in terms of general changes I would suggest not allowing people to upgrade from submarine to battleship while underwater. The reason for this is because there have been many instances in which I am in a 1 vs 1 battle against another sub, and even though I have more skill than my opponent, they end up killing me by getting close and ramming me underwater with a battleship. I don’t understand how this could work because in real life battleships couldn’t withstand the great pressures from the water at hundreds of feet below the surface. This isn’t realistic and lets people with barely any skill kill experienced players that have lots of skill and might be going for high scores. You should be forced to surface to upgrade to a surface ship. 

Another suggestion that I have is about the dreadnought. In real life the dreadnoughts had 4 unguided torpedoes and so I was wondering why they didn’t have any in the game. As it is now dreadnoughts have no submarine defense which makes them very vulnerable. If they had 4 torpedoes (like they should) it wouldn’t make them too powerful, but would at least give them a chance. Something else is that the Bismarck battleship has 2 vought kingfisher planes. This makes no sense because the kingfisher was an american plane and so never served on any german ships. The Bismarck did have planes but they were the Arado Ar 196. I mean it’s not a major issue, but it's something that you might want to consider changing. One major issue is that it is very hard for small lvl 1 and other low lvl ships to not get killed by larger vessels. There is too much spawn killing that is happening and so I suggest that you create a new biome in the South. I heard about this idea from a few other people, and I really like it. Just like how you have the arctic biome in the North, you could have a swamp, or a lot of small islands with channels, sort of like a littoral zone, that will only allow small ships to navigate through the tight twists and turns. You could have a few lakes and stuff so that it isn’t completely small islands, but they could be surrounded by small islands to keep large ships from getting there. 

Something that would be really cool would be if you added a function that would allow you request a battle with another player. If you send a request and they accept it, you would both be sent to an arena in which you could battle each other in a 1 vs 1 without having to worry about bots or other players interfering. 1 extra thing is that when you kill someone on the current leaderboard (top 10) there could be a quick message that appears on the screen that says they’ve been killed and by who (sort of like in other io games like diep.io, hole.io etc). This faint message would avoid people lying about who killed them. Sometimes when you kill someone that is a high level and then they say “you didn’t kill me, some other guy did”. This would avoid this conflict and allow people to actually know who did what. This isn’t really important, but something to consider. Another suggestion is that after trying this ship many times, I have come to the conclusion that the Zumwalt destroyer is a very good surface combatant, but lacks anti-submarine capabilities. Seeing that it is at such a high lvl (8), I think that it should have 1 extra weapon. Just like in real life, the Zumwalt destroyer should have a towable sonar decoy (AN/SLQ-25 Nixie). Basically this is a sonar decoy that is towed out the back of the ship and is made to mimic the noise created by the ships engines. By doing this the Zumwalt would be better equipped to defend against underwater threats. Also, what kind of planes can’t fly over land? To me this makes absolutely no sense. Planes should be able to fly over land because in real life I don’t know of any planes that can’t do this.

I also agree with what some people have been saying about diesel subs. Diesel submarines should have limited time underwater before they must surface not only for air, but to run their diesel engines to charge their batteries. Nuclear subs don’t have to do this because they can spend months underwater while diesel can’t. One other thing that I would suggest is that when a member of a fleet leaves that fleet (or gets kicked) they and the fleet are immune to each other's weapons for about 5 to 10 seconds. This would reduce the amount of traitors by allowing the fleet to prepare for it.  I would slightly upgrade the moskva by giving it 2 guns because in real life it does have two 57mm guns. I really think that it needs this because its only weapon are torpedoes (helicopters carry them as well) and so needs something that it can use to fight off against small ships that are too fast for torpedoes. I would also suggest a slight change in the all time leaderboard. Beside the high score you should put the number of kills that player had during their top score run. This would truly tell the skill of the player because with just the score, you aren’t able to tell if they won battles to get that score or if most of it was donated. For example, you might have someone with 50k, but only 40 kills and have someone with 30k but over 50 kills. Even though the person with 50k has more xp, the person with 30k is a better player because they won more battles. This isn’t very necessary, but would help tell who truly is the best player and it’s relatively simple to add some sort of a “kill count”. 

I do know that this is a very long message, and am sorry about this but I didn’t want to have to write lots of small messages and just wanted to let you know of all my thoughts. Overall this is a very good game, probably my personal favorite, and I really appreciate you guys spending all this time creating this game. I know it must have taken you guys a long time to make it and just wanted to tell you how much I really enjoy playing this game. The concept and ideas used in this game seem very original and interesting. I don’t want to criticize this game too much and know that you're all very busy with school and work and all that, but just wanted to let you know what I think. Just so you know, my player name is crusader. Thanks :)

_Originally posted by Crusader48 in https://github.com/SoftbearStudios/mk48/discussions/132#discussioncomment-2857671_

**Comment by finnbear** [comment]
> I have suggested many things before by talking to the server in the mk48.io game, but this seems like a better place to mention them.

Yep! I've gone ahead and moved your feedback to a new issue because, while you do suggest some ships, you have so many other great suggestions :)

> There are multiple ships that I would like to see added to the game. These include the Skjold class corvette, the Nimitz or Gerald R Ford carrier, the Typhoon submarine, the USS Iowa (modernized), a minesweeper such as the HMCS Kingston, a hospital ship, an electronic countermeasures ship (ECM) and maybe a fishing trawler that catches fish to get xp.

- We considered the Skjold but choose to add the Visby instead, because the Skjold lacks torpedoes and aircraft (so would be completely useless against subs)
- A supercarrier like Nimitz or G. R. F. is a likely addition for level 10 :)
- The Typhoon was planned, but is now very much on hold due to current events surrounding its national origin. We might add a different SSBN instead
- The Iowa (Modernized) is almost definitely going to be the level 10 battleship :)
- We would add a minesweeper if mines were a bigger part of the game. Right now, the Lublin minelayer is a niche ship so mines are rarely a problem.
- We are considering adding some sort of repair/hospital ship to heal other ships
- Civilian ships (fishing trawler, passenger ferry, cargo ships) are a possibility, but we would need to solve balancing and ethical issues first

> I would suggest not allowing people to upgrade from submarine to battleship while underwater. The reason for this is because there have been many instances in which I am in a 1 vs 1 battle against another sub, and even though I have more skill than my opponent, they end up killing me by getting close and ramming me underwater with a battleship. I don’t understand how this could work because in real life battleships couldn’t withstand the great pressures from the water at hundreds of feet below the surface. This isn’t realistic and lets people with barely any skill kill experienced players that have lots of skill and might be going for high scores. You should be forced to surface to upgrade to a surface ship.

This is an intentional feature/easter egg, and we are unlikely to remove it entirely. However, we absolutely don't want it to be abused in the way you describe, so we might make it so submerged surface ships don't collide with submerged submarines.

> nother suggestion that I have is about the dreadnought. In real life the dreadnoughts had 4 unguided torpedoes and so I was wondering why they didn’t have any in the game. As it is now dreadnoughts have no submarine defense which makes them very vulnerable. If they had 4 torpedoes (like they should) it wouldn’t make them too powerful, but would at least give them a chance.

This was a mistake/oversight on our end. Thanks for bringing it up!

> Something else is that the Bismarck battleship has 2 vought kingfisher planes. This makes no sense because the kingfisher was an american plane and so never served on any german ships. The Bismarck did have planes but they were the Arado Ar 196. I mean it’s not a major issue, but it's something that you might want to consider changing.

This is just me being too lazy to model more types of planes...but you're right!

> One major issue is that it is very hard for small lvl 1 and other low lvl ships to not get killed by larger vessels. There is too much spawn killing that is happening

Yes, this is definitely the biggest problem with the game right now! (based on negative feedback we receive)

> and so I suggest that you create a new biome in the South. I heard about this idea from a few other people, and I really like it. Just like how you have the arctic biome in the North, you could have a swamp, or a lot of small islands with channels, sort of like a littoral zone, that will only allow small ships to navigate through the tight twists and turns. You could have a few lakes and stuff so that it isn’t completely small islands, but they could be surrounded by small islands to keep large ships from getting there.

In the next update, we will fix the spawning algorithm to avoid creating a bloodbath south of the arctic biome border. We also plan to add the biome you suggest (shallow water, tightly packed islands).

> Something that would be really cool would be if you added a function that would allow you request a battle with another player. If you send a request and they accept it, you would both be sent to an arena in which you could battle each other in a 1 vs 1 without having to worry about bots or other players interfering.

I've added this as a [game mode suggestion](https://github.com/SoftbearStudios/mk48/discussions/163)!

> 1 extra thing is that when you kill someone on the current leaderboard (top 10) there could be a quick message that appears on the screen that says they’ve been killed and by who (sort of like in other io games like diep.io, hole.io etc). This faint message would avoid people lying about who killed them. Sometimes when you kill someone that is a high level and then they say “you didn’t kill me, some other guy did”. This would avoid this conflict and allow people to actually know who did what. This isn’t really important, but something to consider.

We plan to solve this with a [kill feed](https://github.com/SoftbearStudios/mk48/issues/65)

> Another suggestion is that after trying this ship many times, I have come to the conclusion that the Zumwalt destroyer is a very good surface combatant, but lacks anti-submarine capabilities

We are improving rocket-torpedo guidance for the next update to help with this.

> the Zumwalt destroyer should have a towable sonar decoy

Towable decoys are really cool but the game doesn't currently support them. They would require the long-distance interaction of two separate "entities," the ship and the decoy.

> Also, what kind of planes can’t fly over land? To me this makes absolutely no sense. Planes should be able to fly over land because in real life I don’t know of any planes that can’t do this.

Planes can fly over some land, just not "bulletproof-grass," an intentional feature that makes dredger bases immune from air, rocket, shell, and missile attack (which would make them pretty useless).

> I also agree with what some people have been saying about diesel subs. Diesel submarines should have limited time underwater before they must surface not only for air, but to run their diesel engines to charge their batteries. 

We considered such a mechanic, and while it is realistic, it is low priority until we see the game-play problem it solves. Right now, we think diesel subs are under-powered (due to their slow speed), so making them worse doesn't seem to solve any pressing issue.

> One other thing that I would suggest is that when a member of a fleet leaves that fleet (or gets kicked) they and the fleet are immune to each other's weapons for about 5 to 10 seconds. This would reduce the amount of traitors by allowing the fleet to prepare for it.

This gets suggested a lot, and for good reason. It does add some complexity, though, as the game might have to store an unbounded collection of recent team memberships.

> I would slightly upgrade the moskva by giving it 2 guns because in real life it does have two 57mm guns. I really think that it needs this because its only weapon are torpedoes (helicopters carry them as well) and so needs something that it can use to fight off against small ships that are too fast for torpedoes.

Those guns are currently regarded as [secondary weapons](https://github.com/SoftbearStudios/mk48/issues/72), but maybe they are high enough caliber to be primary weapons. Hmmmm...

> I would also suggest a slight change in the all time leaderboard. Beside the high score you should put the number of kills that player had during their top score run. This would truly tell the skill of the player because with just the score, you aren’t able to tell if they won battles to get that score or if most of it was donated. For example, you might have someone with 50k, but only 40 kills and have someone with 30k but over 50 kills. Even though the person with 50k has more xp, the person with 30k is a better player because they won more battles. This isn’t very necessary, but would help tell who truly is the best player and it’s relatively simple to add some sort of a “kill count”.

This is an interesting idea. One technical issue is that our leaderboard is managed by our custom game engine, which we plan to use for future games. It would be a bit problematic to add mk48.io-specific features, and idk if "kills" will be a central mechanic in *all* future games.

> I do know that this is a very long message, and am sorry about this but I didn’t want to have to write lots of small messages and just wanted to let you know of all my thoughts.

Totally fine. We appreciate your feedback!

> Overall this is a very good game, probably my personal favorite, and I really appreciate you guys spending all this time creating this game. I know it must have taken you guys a long time to make it and just wanted to tell you how much I really enjoy playing this game. The concept and ideas used in this game seem very original and interesting. I don’t want to criticize this game too much and know that you're all very busy with school and work and all that, but just wanted to let you know what I think. Just so you know, my player name is crusader. Thanks :)

Thank you for your compliments, suggestions, and for being very understanding of our current situation!

@Crusader48

**Comment by WallKey1** [comment]
"...and so I suggest that you create a new biome in the South. I heard about this idea from a few other people, and I really like it. Just like how you have the arctic biome in the North, you could have a swamp, or a lot of small islands with channels, sort of like a littoral zone, that will only allow small ships to navigate through the tight twists and turns. You could have a few lakes and stuff so that it isn’t completely small islands, but they could be surrounded by small islands to keep large ships from getting there."

"In the next update, we will fix the spawning algorithm to avoid creating a bloodbath south of the arctic biome border. We also plan to add the biome you suggest (shallow water, tightly packed islands). "

**(Definitely not me exploiting the new biome and killing all who spawn there)** 😳💀

I wouldn't let a ongoing war since 20 feb 2014 prevent adding a submarine to an I.O game. 

 Also filter blocks people from using Hitler but not Communist leaders like Stalin who killed 9 million people🤷🏻‍♀️ Even tho get get past it by getting creative with symbols etc. 

Noticed the delay in submarine submerging more realistic but means I had lots of fun blitzing them with shells before they can sink away. Is that an issue, IDK maybe if you are a die hard submarine fan and you keep getting blown up when surfacing for xp. 

Great stuff, loved the updates / trial server👍🏻. 

![JP Ship -](https://user-images.githubusercontent.com/100443626/171849157-ce0b6c37-38c0-4f5d-9e60-044bc369cffe.PNG)
![Slug1 -](https://user-images.githubusercontent.com/100443626/171849166-e7a70540-4027-4f57-a135-4452aab1d0b8.PNG)
![Slug2 -](https://user-images.githubusercontent.com/100443626/171849170-cc3b20e2-b043-4861-99a6-c0374249fc44.PNG)

**Comment by Crusader48** [comment]
How did you get the Iowa and the small islands?

**Comment by WallKey1** [comment]
> How did you get the Iowa and the small islands?

Ima Parking expert

**Comment by WallKey1** [comment]
> How did you get the Iowa and the small islands? 

Unless you mean why is there islands and Iowa, well check the servers... :D

**Comment by Crusader48** [comment]
is there a different server with the Iowa on it? And 16 missiles with 9 shells? that seems a little too op.

**Comment by finnbear** [comment]
> is there a different server with the Iowa on it? And 16 missiles with 9 shells? that seems a little too op.

All currently running servers have the update at this point. Some players think the Iowa is underpowered and some think it is overpowered. I will wait until (either) problem becomes more obvious ;)

**Comment by Crusader48** [comment]
There is an Iowa? COOL

**Comment by WallKey1** [comment]
I have tested upgrading under large ships and I have been getting killed much more easily, so this seems to be working well! Going to test upgrading to kill subs next. For example upgrading from Akula to Bismarak etc essentially poking people to death. 

Iowa is great - 3 helis 👍🏿 

Islands could be closer together with more grass when you do the bug fix, would prevent me from getting inside with fat ships or too many missiles from enternering from outside the biome.  However it is a great death trap for submarines.

**Comment by Crusader48** [comment]
I also agree. This new update is really good. I love the Iowa. At first I was wondering why it had so many missiles, but since you need about 2300 points to get it, it's a good amount of weapons. After using it for a while, I have noticed that it isn't too powerful but still really good. It's a perfect balance of anti-ship and anti-sub weapons as well as not having too much health. The small islands are also really good but I am still able to maneuver between the islands with a massive battleship.  And as you said it does do a great job at keeping out subs. The delay for subs when submerging is also good because it keeps them from popping up, taking xp from a kill and re-submerging. Very well done :)

**Comment by WallKey1** [comment]
Will keep an eye out for future update but won't be playing for now, I love playing submarines now they are painful to use. Even tho I agree with them taking longer to sink as its realistic I don't like using them at all anymore😂 

After some testing I think Kirov is OG again, before it was closely balanced between seawolf, Kirov and Yas.

Kirov being the great allrounder 
Seawolf being the Yas killer/sub destroyer
And Yas being the Ship destroyer  

Maybe add Admiral Nakhimov (Updated Kirov) 2023 for LV 10+ suggestions. 

**Also the Standard Kirov carries at least 200 sams yet in the game the Type 55 has more loaded at any one time?** 

 Iowa seems quite easy to kill with Kirov bringing its health down by around 70% outside its shell range. 

cya
